### PR TITLE
feature/arrays + FLAT

### PR DESCRIPTION
**Pull request by Pauatro**

### CODE REVIEW

File: staff/pau-atienza/arrays/array-functions/every.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+    }
+    var counter = 0;
+    for (var i = 0; i < array.length; i++) {
+        if(expression(array[i], i, array)){
+            counter +=1;
+        };
+    }
+    if(counter === array.length){
+        return true;
+    }
```
Comment (Aitor1111) [inline staff/pau-atienza/arrays/array-functions/every.js:13]: You can do:
``` if ( !expression(array[i], i, array) ) return false```
Then you can save the ```var counter```and the if comparator from line 11 up to 13.

File: staff/pau-atienza/arrays/array-functions/every.spec.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
+describe('every', function () {
+    it('returns true if every element in the provided array satisfies the provided testing function' , function () {
+        var array = [55, 12, 98, 130, 44];
+        var result = every(array, element => element > 10);
+
+        expect(result).toBe(true);
+    });
+    it('returns false if any elements in the provided array do not satisfy the provided testing function' , function () {
+        var array = [5, 12, 8, 130, 44];
+        var result = every(array, element => element > 6);
```
Comment (Aitor1111) [inline staff/pau-atienza/arrays/array-functions/every.spec.js:10]: That ES6...

File: staff/pau-atienza/arrays/array-functions/filter.js [hunk 3]
```diff
@@ -0,0 +1,11 @@
+
+function filter(array, expression) {
+    var result = [];
+
+    for (let i = 0; i < array.length; i++) {
```
Comment (Aitor1111) [inline staff/pau-atienza/arrays/array-functions/filter.js:5]: I'm confused, that looks like a ```let``` isn't it?

File: staff/pau-atienza/arrays/array-functions/pop.spec.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
+describe('The pop method ', function () {
+    it('should return undefined when the array is empty', function () {
+        var array = [];
+
+        var deletedValue = pop(array);
```
Comment (Aitor1111) [inline staff/pau-atienza/arrays/array-functions/pop.spec.js:5]: also a de```let```ed...

File: staff/pau-atienza/arrays/array-functions/filter.js (hunk 3 above)
Comment (CaliZam) [reply staff/pau-atienza/arrays/array-functions/filter.js:5]: ahhh he did'nt finished was Imcomp- let -e

File: staff/pau-atienza/arrays/array-functions/every.js (hunk 1 above)
Comment (Pauatro) [reply staff/pau-atienza/arrays/array-functions/every.js]: Mucha razón, lo he cambiado :)

**Comment by lichab** [comment]
Hey Pau, great job!. I will merge this but will give you the same feedbacj than Hector haha :laughing: , remember that using the try catch inside the tests is supposed to be replaced with the expect(function).toThrow(), just keep it in mind because the try catch is gonna end up being used in tests but in different situations. +1

**Pull request merged**